### PR TITLE
[GSW-ALL] fix: Fix entire comments

### PR DESCRIPTION
**Pull request by khiemldk**

### CODE REVIEW

File: packages/web/src/components/common/bar-graph/BarGraph.tsx [hunk 1]
```diff
@@ -334,7 +329,7 @@ const BarGraph: React.FC<BarGraphProps> = ({
               <stop offset="100%" stopColor={redColor.end} />
             </linearGradient>
           </defs>
-          {radiusBorder &&
+          {/* {radiusBorder &&
```
Comment (jinoosss) [inline packages/web/src/components/common/bar-graph/BarGraph.tsx:332]: will leave this unused code?

File: packages/web/src/components/common/exchange-rate/ExchangeRate.tsx [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  const scientificNotation = parseFloat(decimalFraction).toExponential();
+  const exponent = parseFloat(scientificNotation.split("e")[1]);
+  return Math.abs(exponent);
+}
+
+const ExchangeRate: React.FC<Props> = ({ value }) => {
+  const temp = `${value}`;
+  const numberOfZero = countZeros(value);
+  if (Number(value) === 0) return <>0</>;
+  if (value[0] !== "0" || !value.startsWith("0.00000")) return <>{removeTrailingZeros(value)}</>;
```
Comment (jinoosss) [inline packages/web/src/components/common/exchange-rate/ExchangeRate.tsx:18]: Is it right to compare strings ?
How about a number comparison?

File: packages/web/src/components/common/pool-graph/PoolGraph.tsx [hunk 3]
```diff
@@ -506,12 +506,12 @@ export const PoolGraphBinTooptip: React.FC<PoolGraphBinTooptipProps> = ({
           </span>
           <span className="amount total-amount">
             <MissingLogo symbol={tooltipInfo.tokenA.symbol} url={tooltipInfo.tokenA.logoURI} className="logo" width={20} mobileWidth={20} />
-            <span className="hidden">{tooltipInfo.tokenAAmount || "0"}</span>
+            <span className="hidden" style={(tooltipInfo.tokenAAmount || "0").length > 21 ? { fontSize: "12px"} : {}}>{tooltipInfo.tokenAAmount || "0"}</span>
```
Comment (jinoosss) [inline packages/web/src/components/common/pool-graph/PoolGraph.tsx:509]: I think it would be better to separate the class name rather than writing it in the component style attribute, what do you think?

File: packages/web/src/components/common/pool-graph/PoolGraph.tsx [hunk 4]
```diff
@@ -506,12 +506,12 @@ export const PoolGraphBinTooptip: React.FC<PoolGraphBinTooptipProps> = ({
...
           </span>
-          {isPosition && !tooltipInfo.isBlackBar ? <span className="amount mr-3">
+          {isPosition && !tooltipInfo.isBlackBar ? <span className="amount w-100">
             <span className="hidden">{tooltipInfo.myTokenAAmount || "0"}</span>
           </span> : ""}
-          <span className="price-range" style={tokenAPriceRangeStr.length > 21 ? { fontSize: "12px"} : {}}>{tokenAPriceRangeStr}</span>
+          <span className="price-range" style={(tokenAPriceRangeStr.length || "0") > 21 ? { fontSize: "12px"} : {}}>{tokenAPriceRangeStr}</span>
```
Comment (jinoosss) [inline packages/web/src/components/common/pool-graph/PoolGraph.tsx:514]: I think it would be better to separate the class name rather than writing it in the component style attribute, 
what do you think?

File: packages/web/src/components/common/select-price-range-cutom-controller/SelectPriceRangeCutomController.tsx [hunk 5]
```diff
@@ -1,31 +1,44 @@
...
-  feeTier: SwapFeeTierType
+  feeTier: SwapFeeTierType;
   tickSpacing?: number;
   selectedFullRange: boolean;
   onSelectCustomRange: () => void;
   changePrice: (price: number) => void;
   decrease: () => void;
   increase: () => void;
-  currentPriceStr: string;
+  currentPriceStr: any;
```
Comment (jinoosss) [inline packages/web/src/components/common/select-price-range-cutom-controller/SelectPriceRangeCutomController.tsx:34]: Why is this type `any`?

File: packages/web/src/components/common/select-price-range-cutom-controller/SelectPriceRangeCutomController.tsx [hunk 6]
```diff
@@ -124,35 +147,91 @@ const SelectPriceRangeCutomController: React.FC<SelectPriceRangeCutomControllerP
...
+      ? subscriptFormat(value)
+      : value === "∞"
+      ? value
+      : convertToKMB(Number(value).toFixed(4), 4);
+  const priceValueString = (
+    <>
+      1 {token0Symbol} =&nbsp;{exchangePrice}&nbsp;{token1Symbol}
+    </>
+  );
+  console.log(currentValue, "123123");
```
Comment (jinoosss) [inline packages/web/src/components/common/select-price-range-cutom-controller/SelectPriceRangeCutomController.tsx:180]: Remove the console log.

File: packages/web/src/components/home/card-list/CardList.tsx [hunk 7]
```diff
@@ -2,44 +2,69 @@ import IconTriangleArrowUp from "@components/common/icons/IconTriangleArrowUp";
...
 }
 
-function isTokenInfo(info: CardListTokenInfo | CardListPoolInfo): info is CardListTokenInfo {
+const enum CardType {
+  "keyStats",
+  "token",
+  "pool",
+}
+
+function typeTokenInfo(
```
Comment (jinoosss) [inline packages/web/src/components/home/card-list/CardList.tsx:26]: The function name looks like a variable name. What if we change it to a verb?

File: packages/web/src/components/home/card-list/CardList.tsx [hunk 8]
```diff
@@ -2,44 +2,69 @@ import IconTriangleArrowUp from "@components/common/icons/IconTriangleArrowUp";
...
   return (
     <CardListWrapper>
       {list.map((item, index) =>
-        isTokenInfo(item) ? (
+        typeTokenInfo(item) === CardType.token ? (
           <CardListTokenItem
             key={index}
             index={index + 1}
-            item={item}
+            item={item as CardListTokenInfo}
```
Comment (jinoosss) [inline packages/web/src/components/home/card-list/CardList.tsx:50]: Why did you remove type guard and use type assertion?
https://github.com/gnoswap-labs/gnoswap-interface/pull/316/files#diff-d9c968003505678c20746c58eeccb26e94d3c70a0d51758a806051405b80302cL16

File: packages/web/src/hooks/swap/use-swap-handler.tsx [hunk 9]
```diff
@@ -853,10 +855,10 @@ export const useSwapHandler = () => {
     type,
     tokenB?.symbol,
   ]);
-
+  
   useEffect(() => {
     if (!tokenA || !tokenB) {
-      setIsLoading(false);
+      // setIsLoading(false);
```
Comment (jinoosss) [inline packages/web/src/hooks/swap/use-swap-handler.tsx:861]: Remove unused line.

**Review by jinoosss — COMMENTED** [review]
Thank you for your work.
I left a few comments.

File: packages/web/src/components/home/card-list/CardList.tsx (hunk 8 above)
Comment (khiemldk) [reply packages/web/src/components/home/card-list/CardList.tsx:50]: now we have 3 option for this

File: packages/web/src/components/common/bar-graph/BarGraph.tsx (hunk 1 above)
Comment (khiemldk) [reply packages/web/src/components/common/bar-graph/BarGraph.tsx]: DONE

File: packages/web/src/components/common/pool-graph/PoolGraph.tsx (hunk 4 above)
Comment (khiemldk) [reply packages/web/src/components/common/pool-graph/PoolGraph.tsx]: DONE

File: packages/web/src/components/common/select-price-range-cutom-controller/SelectPriceRangeCutomController.tsx (hunk 6 above)
Comment (khiemldk) [reply packages/web/src/components/common/select-price-range-cutom-controller/SelectPriceRangeCutomController.tsx]: DONE

File: packages/web/src/components/common/pool-graph/PoolGraph.tsx (hunk 3 above)
Comment (khiemldk) [reply packages/web/src/components/common/pool-graph/PoolGraph.tsx]: DONE

File: packages/web/src/components/home/card-list/CardList.tsx (hunk 7 above)
Comment (khiemldk) [reply packages/web/src/components/home/card-list/CardList.tsx]: DONE

File: packages/web/src/components/home/card-list/CardList.tsx (hunk 8 above)
Comment (jinoosss) [reply packages/web/src/components/home/card-list/CardList.tsx:50]: You can declare a Wrapper component like this.
The Wrapper component just handles branching by type.

It seems better to explicitly branch by typeguard rather than type assertion (I think it's a matter of branching policy management).

#### CardList Component
```
const CardList: React.FC<CardListProps> = ({
  list,
  onClickItem,
  isHiddenIndex = false,
}) => {
  return (
    <CardListWrapper>
      {list.map((item, index) => (
        <CardListItem
          key={index}
          index={index + 1}
          item={item}
          onClickItem={onClickItem}
          isHiddenIndex={isHiddenIndex}
        />
      ))}
    </CardListWrapper>
  );
};
```

#### CardListItem Component
```
const CardListItem: React.FC<CardListItemProps> = ({
  index,
  item,
  isHiddenIndex,
  onClickItem,
}) => {
  function isTokenCardItem(
    item: CardListTokenInfo | CardListPoolInfo | CardListKeyStats,
  ): item is CardListTokenInfo {
    return "token" in item;
  }

  function isKeyStatsCardItem(
    item: CardListTokenInfo | CardListPoolInfo | CardListKeyStats,
  ): item is CardListKeyStats {
    return "label" in item;
  }

  if (isTokenCardItem(item)) {
    return (
      <CardListTokenItem
        index={index}
        item={item}
        isHiddenIndex={isHiddenIndex}
        onClickItem={onClickItem}
      />
    );
  }

  if (isKeyStatsCardItem(item)) {
    return <CardListKeyStatsItem item={item} />;
  }

  return (
    <CardListPoolItem index={index} item={item} onClickItem={onClickItem} />
  );
};
```

Comment (khiemldk) [reply packages/web/src/components/home/card-list/CardList.tsx]: Done

**Review by jinoosss — APPROVED** [review]
LGTM !
